trade-finance-manager-api/src/v1/middleware/validate-sso-feature-flag.test.ts

```diff
@@ -0,0 +1,87 @@
+import { isTfmSsoFeatureFlagEnabled } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { RequestHandler, Request, Response, NextFunction } from 'express';
+import httpMocks, { MockRequest, MockResponse } from 'node-mocks-http';
+import { validateSsoFeatureFlagIsOff, validateSsoFeatureFlagIsOn } from './validate-sso-feature-flag';
+
+jest.mock('@ukef/dtfs2-common', () => ({
+  isTfmSsoFeatureFlagEnabled: jest.fn(),
+}));
+
+describe('validateSsoFeatureFlag', () => {
+  let req: MockRequest<Request>;
+  let res: MockResponse<Response>;
+  let next: jest.Mock | NextFunction;
+
+  beforeEach(() => {
+    ({ req, res } = httpMocks.createMocks());
+    next = jest.fn();
+
+    jest.resetAllMocks();
+  });
+
+  describe('validateSsoFeatureFlagIsOn', () => {
+    const makeRequest = validateSsoFeatureFlagIsOn;
+
+    describe('when the SSO feature flag is disabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(false);
+      });
+
+      itReturnsAnErrorResponse({ makeRequest });
+    });
+
+    describe('when the SSO feature flag is enabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(true);
+      });
+
+      itCallsTheNextMiddleware({
+        makeRequest,
+      });
+    });
+  });
+
+  describe('validateSsoFeatureFlagIsOff', () => {
+    const makeRequest = validateSsoFeatureFlagIsOff;
+
+    describe('when the SSO feature flag is disabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(false);
+      });
+
+      itCallsTheNextMiddleware({
+        makeRequest,
+      });
+    });
+
+    describe('when the SSO feature flag is enabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(true);
+      });
+
+      itReturnsAnErrorResponse({ makeRequest });
+    });
+  });
+
+  function mockIsTfmSsoFeatureFlag(value: boolean) {
+    jest.mocked(isTfmSsoFeatureFlagEnabled).mockReturnValue(value);
+  }
+
+  function itCallsTheNextMiddleware({ makeRequest }: { makeRequest: RequestHandler }) {
```

Comment:
I find this `itReturnsAnErrorResponse` harder to follow - I know it's to be DRY but I think the trade off is readability, I think it'd be clearer to have `expect(next).toHaveBeenCalled();` (same with the error assertions), repeated in the tests themselves. If there was a lot more instances I would prefer the current approach

Can we add `toHaveBeenCalledTimes(1);` for the "next" assertions please?